### PR TITLE
Update language-go.md

### PR DESCRIPTION
**Pull request by thomasr8**
# Description
replaced references to legacy images to next-gen images

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

### CODE REVIEW

File: jekyll/_cci2/language-go.md [hunk 1]
```diff
@@ -39,12 +39,12 @@ jobs: # basic units of work in a run
   build: # runs not using Workflows must have a `build` job as entry point
     docker: # run the steps with Docker
       # CircleCI Go images available at: https://hub.docker.com/r/circleci/golang/
-      - image: circleci/golang:1.16
+      - image: cimg/golang:1.16
```
Comment (rosieyohannan) [inline jekyll/_cci2/language-go.md:42]: See https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034#deprecated-images-3 `golang` -> `go`

File: jekyll/_cci2/language-go.md [hunk 2]
```diff
@@ -39,12 +39,12 @@ jobs: # basic units of work in a run
...
         auth:
           username: mydockerhub-user
           password: $DOCKERHUB_PASSWORD  # context / project UI env-var reference
       # CircleCI PostgreSQL images available at: https://hub.docker.com/r/circleci/postgres/
-      - image: circleci/postgres:9.6-alpine
+      - image: cimg/postgres:9.6-alpine
```
Comment (felicianotech) [inline jekyll/_cci2/language-go.md:47]: -alpine needs to go.

File: jekyll/_cci2/language-go.md [hunk 3]
```diff
@@ -157,12 +157,12 @@ jobs: # basic units of work in a run
...
```
Comment (felicianotech) [inline jekyll/_cci2/language-go.md:165]: -alpine needs to go.

**Review by felicianotech — CHANGES_REQUESTED** [review]

File: jekyll/_cci2/language-go.md (hunk 1 above)
Comment (thomasr8) [reply jekyll/_cci2/language-go.md]: fixed

File: jekyll/_cci2/language-go.md (hunk 2 above)
Comment (thomasr8) [reply jekyll/_cci2/language-go.md]: removed

File: jekyll/_cci2/language-go.md (hunk 3 above)
Comment (thomasr8) [reply jekyll/_cci2/language-go.md]: removed

**Review by felicianotech — APPROVED** [review]

**Pull request merged**